app/src/main/java/ca/klapstein/baudit/data/Problem.java

```diff
@@ -75,4 +82,9 @@ public void setDate(Date date) {
     public String getTimeStamp() {
         return getBauditDateFormat().format(date);
     }
-}
+  
+    @Override
+    public int compareTo(Problem problem) {
+        return this.date.compareTo(problem.getDate());
```

Comment:
this is my preference speaking, but usually if it is not needed i avoid using the `this.field` syntax to keep code simple and easier to read.

I only use the `this.field` in constructors or where it is needed due to name collisions.